pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -762,9 +762,6 @@ func (vs *VSphere) InstanceID(ctx context.Context, nodeName k8stypes.NodeName) (
 		}
 		vm, err := vs.getVMFromNodeName(ctx, nodeName)
 		if err != nil {
-			if err == vclib.ErrNoVMFound {
```

Comment:
it suppose to return cloudprovider.InstanceNotFound when vm not found,
but it did double error type translation here with a retry() logic,
after removing the retry() (which is not necessary), the logic is simpler, and easier to understand how double translation happened